### PR TITLE
fp2: Change in Credential Management Approach

### PR DESCRIPTION
**Pull request by hdlee27**
Check all that apply

# Type of Change

- [ ] WWST Certification Request
     - If this is your first time contributing code:
          - [ ] I have reviewed the README.md file
          - [ ] I have reviewed the CODE_OF_CONDUCT.md file
          - [ ] I have signed the CLA
     - [ ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [x] Bug fix
- [ ] New feature
- [ ] Refactor

# Checklist

- [x] I have performed a self-review of my code
- [ ] I have commented my code in hard-to-understand areas
- [x] I have verified my changes by testing with a device or have communicated a plan for testing
- [ ] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change

# Summary of Completed Tests

### CODE REVIEW

File: drivers/Aqara/aqara-presence-sensor/src/discovery.lua [hunk 1]
```diff
@@ -4,47 +4,59 @@ local fields = require "fields"
...
 local socket = require "cosock.socket"
 
--- mapping from device DNI to info needed at discovery/init time
-local device_discovery_cache = {}
+local processing_devices = {}
 
-local function set_device_field(driver, device)
-  local device_cache_value = device_discovery_cache[device.device_network_id]
+function discovery.set_device_field(driver, device)
+  local device_cache_value = driver.datastore.discovery_cache[device.device_network_id]
```
Comment (cjswedes) [inline drivers/Aqara/aqara-presence-sensor/src/discovery.lua:10]: What is the reason for trying to persist the discovery cache in the datastore? Please explain the use case where this is needed and not covered by the local discovery cache.

Comment (hdlee27) [reply drivers/Aqara/aqara-presence-sensor/src/discovery.lua:10]: During Near by Search, it was found that Discovery stopped when user exit the page by pressing Back button.
At this time, drivers without registered devices appear to be killed, and the local cache disappeared when the driver was executed again.
Datastore was used to keep the cache even when running the driver again after the driver was killed.

Comment (cjswedes) [reply drivers/Aqara/aqara-presence-sensor/src/discovery.lua:10]: > it was found that Discovery stopped when user exit the page by pressing Back button.
At this time, drivers without registered devices appear to be killed, and the local cache disappeared when the driver was executed again

This is how the system is intended to work.

It seems like a nice feature for a user who is going in and out of the scan near by search wondering why a device isn't being discovered, but is that something we need to handle? If they are backing out of scan nearby, I assume they want to stop scanning and do not expect to join a device. There is also a small concern that the cache may become outdated with lots of time between scan attempts.

**Review by cjswedes — APPROVED** [review]
This seems like a low risk change, so I am approving. But the optimization it's making to discovery in situations where the user is starting/stopping scan nearby quickly doesn't seem like a use case we need to optimize IMO.

**Pull request merged**